servers/src/main/java/tachyon/worker/block/TieredBlockStore.java

```
@@ -192,15 +203,24 @@ public void abortBlock(long userId, long blockId) throws IOException {
   }
 
   @Override
-  public void requestSpace(long userId, long blockId, long additionalBytes) throws IOException {
+  public void requestSpace(long userId, long blockId, long additionalBytes)
+      throws NotFoundException, OutOfSpaceException {
+    if (additionalBytes < 0) {
+      throw new IllegalArgumentException("additionalBytes = " + additionalBytes
+          + ", is less than 0");
+    }
     // TODO: Change the lock to read lock and only upgrade to write lock if necessary
     mEvictionLock.writeLock().lock();
     try {
       TempBlockMeta tempBlockMeta = mMetaManager.getTempBlockMeta(blockId);
       freeSpaceInternal(userId, additionalBytes, tempBlockMeta.getBlockLocation());
       // Increase the size of this temp block
-      mMetaManager.resizeTempBlockMeta(tempBlockMeta, tempBlockMeta.getBlockSize()
-          + additionalBytes);
+      try {
+        mMetaManager.resizeTempBlockMeta(tempBlockMeta, tempBlockMeta.getBlockSize()
+            + additionalBytes);
+      } catch (InvalidStateException fpe) {
+        // since we have assured additionalBytes >= 0, this exception should not appear
```

I don't think we should swallow this exception.